server/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -48,8 +48,8 @@
  * nested elements can be added via {@link #declareObject(BiConsumer, ContextParser, ParseField)} which should be satisfied where possible
  * by passing another instance of {@link ObjectParser}, this one customized for that Object.
  * <p>
- * This class works well for object that do have a constructor argument or that can be built using information available from earlier in the
- * XContent. For objects that have constructors with required arguments that are specified on the same level as other fields see
+ * This class works well for object that doesn't have a constructor argument or that can be built using information available from earlier
+ * in the XContent. For objects that have constructors with required arguments that are specified on the same level as other fields see
```

Comment:
could you leave this change out please? I don't exactly understand these docs, so maybe we can try and fix them separately and give such change more visibility.

Reply:
Okay, I'll remove it here. I just find this when I read the docs of `ObjectParser/ConstructingObjectParser` :) because I feel like it seems a typo ? To say `ConstructingObjectParser` is to construct object which has constructor with arguments, and `ObjectParser` is rather for object only has a constructor without arguments, so we need to set its fields by other ways (setter, other methods etc.) ?